tracer/src/Datadog.Trace/Iast/ReturnedHeadersAnalyzer.cs

```diff
@@ -35,25 +36,72 @@ internal static void Analyze(NameValueCollection responseHeaders, IntegrationId
     internal static void Analyze(IHeaderDictionary responseHeaders, IntegrationId integrationId, string serviceName, int responseCode, string protocol)
 #endif
     {
-        if (string.IsNullOrEmpty(serviceName) || IsIgnorableResponseCode((HttpStatusCode)responseCode))
+        AnalyzeXContentTypeOptionsVulnerability(responseHeaders, integrationId, serviceName, responseCode);
+        AnalyzeStrictTransportSecurity(responseHeaders, integrationId, serviceName, responseCode, protocol);
+        AnalyzeUnvalidatedRedirect(responseHeaders, integrationId);
+    }
+
+#if NETFRAMEWORK
+    internal static void AnalyzeXContentTypeOptionsVulnerability(NameValueCollection responseHeaders, IntegrationId integrationId, string serviceName, int responseCode)
+#else
+    internal static void AnalyzeXContentTypeOptionsVulnerability(IHeaderDictionary responseHeaders, IntegrationId integrationId, string serviceName, int responseCode)
+#endif
+    {
+        try
         {
-            return;
-        }
+            IastModule.OnExecutedSinkTelemetry(IastInstrumentedSinks.XContentTypeHeaderMissing);
```

Comment:
It looks like this is now executed first, before the serviceName and responseCode check. Is that intentional?

Reply:
Yes. The idea is to know how many times the program goes through this point, independently from the conditions and the outcome